docs/diagnostics/PX1050.md

```diff
@@ -0,0 +1,92 @@
+# PX1050
+This document describes the PX1050 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                                        | Type  | Code Fix    | 
+| ------ | -------------------------------------------------------------------------------------------------------- | ----- | ----------- | 
+| PX1050 | Hardcoded strings cannot be used as parameters for localization methods and `PXException` constructors. | Error | Unavailable |
+
+## Diagnostic Description
+You should use message strings from the classes with the `PX.Common.PXLocalizableAttribute` attribute instead of hardcoded strings in the following cases:
+
+ - In the message parameter of the static localization methods of the `PX.Data.PXLocalizer` class
+ - In the message parameter of the static localization methods of the `PX.Data.PXMessages` class
+ - In the message parameter of the constructor of the `PXException` class and all derived classes
+
+ Hardcoded strings cannot be localized.
+
+ To fix the issue, you move the hardcoded string to a class with the `PX.Common.PXLocalizableAttribute` attribute and replace the hardcoded string with the constants from this class.
```

Comment:
> you move the hardcoded string to a class with the `PX.Common.PXLocalizableAttribute` attribute

Also it is necessary to declare that string as `public const` field in that class.